sdcm/nemesis.py

```diff
@@ -1866,6 +1867,34 @@ def modify_table_bloom_filter_fp_chance(self):  # pylint: disable=invalid-name
         """
         self._modify_table_property(name="bloom_filter_fp_chance", val=random.random() / 2)
 
+    def toggle_table_gc_mode(self):
+        """
+            Alters a non-system table tombstone_gc_mode option.
+            Choose the alternate option of 'repair' / 'timeout'
+             (*) The other available values of 'disabled' / 'immediate' are not tested by
+             this nemesis since not applicable to a longevity test.
+        """
+        all_ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)
+        non_mview_ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node, filter_out_mv=True)
+
+        if not all_ks_cfs:
+            raise UnsupportedNemesis(
+                'Non-system keyspace and table are not found. toggle_table_gc_mode nemesis can\'t run')
+
+        mview_ks_cfs = list(set(all_ks_cfs) - set(non_mview_ks_cfs))
+        keyspace_table = random.choice(all_ks_cfs)
+        keyspace, table = keyspace_table.split('.')
+        if get_gc_mode(node=self.target_node, keyspace=keyspace, table=table) != GcMode.REPAIR:
+            new_gc_mode = GcMode.REPAIR
+        else:
+            new_gc_mode = GcMode.TIMEOUT
+        new_gc_mode_as_dict = {'mode': new_gc_mode.value}
+
+        alter_command_prefix = 'ALTER TABLE ' if keyspace_table not in mview_ks_cfs else 'ALTER MATERIALIZED VIEW '
+        cmd = alter_command_prefix + f" {keyspace_table} WITH tombstone_gc = {new_gc_mode_as_dict};"
+        self.log.info("Alter GC mode query to execute: %s", cmd)
+        self.target_node.run_cqlsh(cmd)
+
```

Comment:
The fact that the nemesis "stop" here, doesn't mean it's done.
There should be some kind of verification that something happened according to expected result or same behaviour in terms of metrics and at the minimum a sleep or wait to make sure it behave sanly before the next nemesis arrives.
I would even consider toggling back.

Reply:
so let's add a repair after this change, and then we can change back to the original configuration (that would be the meaning of toggling, in a nemesis).

and the assumption that functional tests are done somewhere else is not good.. the nemesis shouldn't assume anything

Reply:
The nemesis is expected to have zero impact at short term.
BTW, i think SCT never tests the original gc-grace-period threshold since it's value is more than a week.
only in a very rare case, after we change this default to a much lower value like 1 day, the nemesis might have some immediate effect - tombstones will be deleted on compactions. still not sure it is noticed easily.
and of course the preconditions is that a load with deletions is running in background.
The long term impact of this nemesis would be that after a future nemesis that runs a repair - tombstones are deleted by compactions.

Reply:
ok, make sense.
But maybe we should also decrease the default at least in some longevities that we may have tombstones there.